packages/common/random-access-storage/src/browser/web-fs.ts

```diff
@@ -0,0 +1,173 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import assert from 'node:assert';
+import { EventEmitter } from 'node:events';
+import { callbackify } from 'node:util';
+import { RandomAccessStorage } from 'random-access-storage';
+
+import { synchronized } from '@dxos/async';
+import { log } from '@dxos/log';
+
+import { Directory, File, Storage, StorageType, getFullPath } from '../common';
+
+/**
+ *
+ */
+export class WebFS implements Storage {
+  readonly type = StorageType.WEBFS;
+
+  protected readonly _files = new Map<string, File>();
+  protected _root?: FileSystemDirectoryHandle;
+
+  constructor(private readonly path: string) {}
+
+  public get size() {
+    return this._files.size;
+  }
+
+  private _getFiles(path: string): Map<string, File> {
+    const fullPath = getFullPath(this.path, path);
```

Comment:
is this for directory.list method? this doesn't seem like it would work corretly. this would only show previously opened files, not all of the files on disk. there's a browser api to enum a directory, lets use that instead

Reply:
That is a bug (or expected behavior) in all our implementations for random-access-storage. I added a test that fails for all implementations and TODO in our issue, so it is weird to fix it only for WebFS, let's do it in separate PR for all implementations